plugins/linuxplugin/l3plugin/data_resync.go

```diff
@@ -64,36 +70,236 @@ func (plugin *LinuxRouteConfigurator) Resync(routes []*l3.LinuxStaticRoutes_Rout
 			if err != nil {
-				errs = append(errs, err)
+				return nil, fmt.Errorf("RESYNC Linux route %s: failed to switch to namespace %s: %v",
+					nbRoute.Name, nbRoute.Namespace.Name, err)
 			}
+			defer revertNs()
+		}
+	}
+	var linuxRoutes []netlink.Route
+	// Look for routes using destination IP address
+	if nbRoute.DstIpAddr != "" {
+		_, dstNetIP, err := net.ParseCIDR(nbRoute.DstIpAddr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse destination IP address %s: %v", nbRoute.DstIpAddr, err)
+		}
+		linuxRts, err := netlink.RouteGet(dstNetIP.IP)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read linux route %s using address %s: %v",
+				nbRoute.Name, nbRoute.DstIpAddr, err)
+		}
+		if linuxRts != nil {
+			linuxRoutes = append(linuxRoutes, linuxRts...)
+		}
+	}
+	// Look for routes using interface
+	if nbRoute.Interface != "" {
+		link, err := netlink.LinkByName(nbRoute.Interface)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read interface %s: %v", nbRoute.Interface, err)
+		}
+		linuxRts, err := netlink.RouteList(link, netlink.FAMILY_ALL)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read linux route %s using interface %s: %v",
+				nbRoute.Name, nbRoute.Interface, err)
+		}
+		if linuxRts != nil {
+			linuxRoutes = append(linuxRoutes, linuxRts...)
 		}
 	}
 
-	// Dump pre-existing not managed arp entries
-	err := plugin.LookupLinuxRoutes()
+	if len(linuxRoutes) == 0 {
+		plugin.Log.Debugf("Equivalent for route %s was not found", nbRoute.Name)
+	}
+
+	return linuxRoutes, nil
+}
+
+// Compare all route parameters and returns true if routes are equal, false otherwise
+func (plugin *LinuxRouteConfigurator) isRouteEqual(rtIdx int, nbRoute, linuxRt *l3.LinuxStaticRoutes_Route) bool {
+	// Common fields (interface, gateway)
+	if nbRoute.Interface != linuxRt.Interface {
+		plugin.Log.Debugf("Linux route %d: interface is different (NB: %s, Linux: %s)",
+			rtIdx, nbRoute.Interface, linuxRt.Interface)
+		return false
+	}
+	if nbRoute.GwAddr != linuxRt.GwAddr {
+		plugin.Log.Debugf("Linux route %d: gateway is different (NB: %s, Linux: %s)",
+			rtIdx, nbRoute.GwAddr, linuxRt.GwAddr)
+		return false
+	}
+	// Default route
+	if nbRoute.Default {
+		if !linuxRt.Default {
+			plugin.Log.Debugf("Linux route %d: NB route is default, but linux route is not", rtIdx)
+			return false
+		}
+		if nbRoute.Metric != linuxRt.Metric {
+			plugin.Log.Debugf("Linux route %d: metric is different (NB: %s, Linux: %s)",
+				rtIdx, nbRoute.Metric, linuxRt.Metric)
+			return false
+		}
+		return true
+	}
+	// Static route
+	_, nbIPNet, err := net.ParseCIDR(nbRoute.DstIpAddr)
 	if err != nil {
-		errs = append(errs, err)
+		plugin.Log.Error(err)
+		return false
+	}
+	if nbIPNet.IP.String() != linuxRt.DstIpAddr {
+		plugin.Log.Debugf("Linux route %d: destination address is different (NB: %s, Linux: %s)",
+			rtIdx, nbIPNet.IP.String(), linuxRt.DstIpAddr)
+		return false
+	}
+	if nbRoute.SrcIpAddr != linuxRt.SrcIpAddr {
```

Comment:
How come the `nbRoute.SrcIpAddr` is compared differently than `nbRoute.DstIpAddr` (without using `net.ParseCIDR`)?

Reply:
because if you read route with a destination address, it is returned without a mask (or to be more exact, the mask is always 32). You can try it yourself with "ip route get <dst>"